File: tests/fixtures/ldap_injection/java/BaselineConstantLdap.java
// Baseline: the filter is a compile-time constant; no taint reaches the sink
// and no LDAP_INJECTION finding fires.  Guards the rule against firing on
// safe-by-construction call sites that simply happen to hit a search API.
import javax.naming.directory.DirContext;
import javax.naming.directory.SearchControls;

public class BaselineConstantLdap {
    private DirContext ctx;

    public Object lookup() throws Exception {
        String filter = "(objectClass=person)";
        return ctx.search("ou=people,dc=example,dc=com", filter, new SearchControls());
    }
}

File: tests/fixtures/ldap_injection/python/baseline_constant_ldap.py
# Baseline: filter is a compile-time constant.  No taint reaches `search_s` so
# no LDAP_INJECTION finding fires.
import ldap


def lookup():
    conn = ldap.initialize("ldap://example.com")
    return conn.search_s(
        "ou=people,dc=example,dc=com", ldap.SCOPE_SUBTREE, "(objectClass=person)"
    )
